docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,122 @@
+expected and the installation will continue.
+
+## Motivation
+
+This is important because users may be able to fix he issues that delay the
+installation instead of having to start it over.
+
+### Goals
+
+### Non-Goals
+
+It is not a goal to change the global installation timeout configured via the
+`INSTALLATION_TIMEOUT` environment variable. That is set by default to `24h`
+and at that point the assisted service stops monitoring the cluster.
+
+## Proposal
+
+### User Stories
+
+#### Allow manually fixing a SasS installation
+
+As a user I tried to create a SNO cluster in the SaaS environment. An issue
+prevented one of the cluster operators from reporting success. After one hour
+the timeout expired and the installation was marked as failed. When I found the
+failed install, I was able to easily resolve the issue and get the cluster
+operator to report success. But the service no longer cared; as far as it was
+concerned, the installation failed and was permanently marked as such. It also
+would not give me the kubeadmin password, which is an important feature of the
+install experience and tricky to obtain otherwise. I would like the service
+inform me about the exceeded timeout, but it should give me the kubeadmin
+password (if available at that point) and after my fixes it should continue
+and eventually mark the installation as successful.
+
+#### Allow manually fixing a late binding installation
+
+As a user I tried to install a cluster with late binding featured enabled
+(deleting the cluster will return the hosts to infrastructure environment). The
+installation timed out and the installation was marked as failed. I connected
+to the cluster and manually fixed the issue. The service will still think that
+there is an error in the cluster, and if I try to perform day 2 operations on
+it they will fail. The only option is to delete the cluster and create another
+one that is marked as installed, but that will cause the host to boot from
+discovery ISO. I would like the service to inform me of the exceeded timeout,
+but after my fixes it should continue and eventually mark the installation as
+successful.
+
+### Implementation Details/Notes/Constraints
+
+To be done.
+
+### Risks and Mitigations
+
+To be done.
+
+## Design Details
+
+None, see the implementation details above.
+
+### Open Questions
+
+- Do we also soften the 24h `INSTALLATION_TIMEOUT` timeout?
+
+- Should we collect must gather logs when the installation timeout expires, even
+  if we don't mark it as failed?
+
+### UI Impact
+
+The UI will need to explicitly show to the user that the cluster installation
+is taking longer than expected, and give suggestions on how to proceed. For
+example, we could present a warning message with this text:
+
+> Cluster installation is taking too long
+>
+> It should have completed in approximately 45 minutes, but it took 23 hours
+> and 52 minutes already. Check the logs to find out why or reset the cluster
```

Comment:
Can we provide more specific guidance than just "check the logs"? For example if a cluster-operator is in a bad state, can we just tell the user to look into that specific CO?

Reply:
Also, what will be available to the users to troubleshoot? I guess in certain cases they will have a kubeadmin already, in certain they wont. Is it the case? If yes, can we explain the user why? Anything else we might provide to the user? Maybe a direct link to "something (tm)" on the cluster to check or hints to troubleshoot (e.g we noticed there might be networking issues, this is how you verify) or something like that?

Reply:
This is limited by (1) the information that the service has and (2) creating some expert system based on that information.  Until we can start working on that, at the minimum, we should identify some common error scenarios and instruct the user on how to determine the causes and fix them.

Reply:
@avishayt the numbers in the UI message are always the same? **45** minutes and **23** hours and **52** minutes? Or do we need to calculate it?

Reply:
I suggest we get the exact message by UX and then decide.  In this case, the 23 hours and 52 minutes can be calculated by the timestamp on the cluster.  The 45 minutes would be hardcoded.  But again, let's get the exact message from UX first.